awx/main/tests/unit/analytics/test_collection_interval.py

```diff
@@ -0,0 +1,203 @@
+import logging
+from unittest import mock
+import pytest
+
+from django.conf import settings
+from django.utils.timezone import now, timedelta
+
+from awx.main.analytics.core import calculate_collection_interval
+
+# This is some bullshit.
+logger = logging.getLogger('awx.main.analytics')
+logger.propagate = True
+
+epsilon = timedelta(minutes=1)
+
+
+class TestIntervalWithSinceAndUntil:
+    @pytest.mark.parametrize('gather', [None, 2, 6])
+    def test_ok(self, caplog, gather):
+        _now = now()
+        _prior = _now - timedelta(weeks=gather) if gather is not None else None
+        until = _now
+        since = until - timedelta(weeks=3)
+        settings.AUTOMATION_ANALYTICS_LAST_GATHER = _prior
```

Comment:
This can result in the setting changing in other tests. There is a settings fixture somewhere that can be used that handles this.

Reply:
ok, should be fixed now

Reply:
> This can result in the setting changing in other tests. There is a `settings` fixture that can be used that handles this.

good catch ^